lib/modules/auth/widgets/login.widget.dart

```diff
@@ -0,0 +1,152 @@
+import 'package:flutter/material.dart';
+import 'package:project/config/theme.config.dart';
+import 'package:project/modules/auth/providers/auth.provider.dart';
+import 'package:provider/provider.dart';
+
+// Define a custom Form widget.
+class Login extends StatefulWidget {
+  const Login({Key? key}) : super(key: key);
+
+  @override
+  LoginState createState() {
+    return LoginState();
+  }
+}
+
+class LoginState extends State<Login> {
+  final _formKey = GlobalKey<FormState>();
+
+  final emailController = TextEditingController();
+  final passwordController = TextEditingController();
+
+  //
+
+  @override
+  void dispose() {
+    // Clean up the controller when the widget is removed from the
+    // widget tree.
+    emailController.dispose();
+    passwordController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AuthProvider _authProvider = Provider.of<AuthProvider>(context);
+    void login() {
+      _authProvider
+          .login(emailController.value.text, passwordController.value.text)
+          .whenComplete(() {
+        ScaffoldMessenger.of(context).showSnackBar(
+          SnackBar(
+            backgroundColor: _authProvider.error == null
+                ? Colors.green
```

Comment:
There is no onSuccess option in the themeData so I'm not sure where I should place this color.

Reply:
poti sa iti definesti tu un field nou in ThemeConfig linistit. La obiectu ala oricum o sa ai acces de aproape oriunde cu ThemeConfig.of(context)
Cam aia ar fi ideea, sa sa se defineasca toate culorile si chestiile de genu intr-un singur loc.